src/odemis/gui/win/acquisition.py

```diff
@@ -874,6 +886,21 @@ def _resume_settings(self):
         self.streambar_controller.enable(True)
         self.streambar_controller.resume()
 
+    def _get_zstack_levels(self):
+        """
+        Calculate the zstack levels from the current focus position and zsteps value
+        :returns (list(float) or None) zstack levels for zstack acquisition
+        """
+        focus_value = self._main_data_model.focus.position.value['z']
+        # Clip zsteps value to allowed range
+        zsteps = numpy.clip(self.zstack_steps.current_value, MINZ, MAXZ)
```

Comment:
Do not use `current_value`, it's an internal attribute (with a bad name!). You should use just the standard wxPython `.GetValue()`.

Also, no need to clip, let's trust the GUI.